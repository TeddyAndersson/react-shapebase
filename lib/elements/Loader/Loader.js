import React from 'react';
import _ from 'lodash';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { spinner } from '../../styles/keyframes'

const Component = (props) => {
  const {
    className,
    ...other
  } = props;

  return (
    <span className={className} {...other}/>
  );
};

Component.propTypes = {

  /**
    * Additonal classes.
    */
  className: PropTypes.string,
  
  /**
    * Color of the paragraph content.
    */
  color: PropTypes.string,

  /**
    * The size of the paragraph.
    */
  size: PropTypes.oneOf([1, 2, 3, 4, 5, 6]),

};

Component.defaultProps = {
    size: 1,
    color: "#000",
} 


/** Styling the Component with styled-components */
const Loader = styled(Component)`
  :before {
    content: '';
    box-sizing: border-box;
    position: absolute;
    width: 20px;
    height: 20px;
    margin-top: -10px;
    margin-left: -10px;
    border-radius: 50%;
    border-top: 2px solid ${props => props.color ? props.color : "#000"};
    opacity: 0.8;
    border-right: 2px solid transparent;
    animation: ${spinner} .6s linear infinite;
  }
`;


export default Loader;