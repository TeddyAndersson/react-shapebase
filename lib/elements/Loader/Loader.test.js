import React from 'react';
import Loader from './Loader'
import renderer from 'react-test-renderer';

test('should render Loader', () => {
  const component = renderer.create(
    <Loader />
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
})
