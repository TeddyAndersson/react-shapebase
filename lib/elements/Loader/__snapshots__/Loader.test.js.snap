// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Loader 1`] = `
<span
  className="sc-bdVaJa bSzbyX"
  color="#000"
  size={1}
/>
`;
